apiserver/facades/controller/caasmodeloperator/operator.go

```diff
@@ -116,16 +116,31 @@ func (a *API) ModelOperatorProvisioningInfo() (params.ModelOperatorInfo, error)
 		return result, errors.Annotate(err, "getting api addresses")
 	}
 
-	registryPath, err := podcfg.GetJujuOCIImagePath(controllerConf, vers)
+	imageRepo, exists := modelConfig.CAASImageRepo()
+	if !exists {
+		imageRepo = controllerConf.CAASImageRepo()
+
+		if imageRepo == "" {
+			imageRepo = podcfg.JujudOCINamespace
+		}
+
+		if err := model.UpdateModelConfig(map[string]interface{}{config.ModelCAASImageRepo: imageRepo}, nil); err != nil {
```

Comment:
Not sure what this is for? We shouldn't be doing a side effect like this in a method that is just supposed to read and return info.

Reply:
Hi ian, this is to updated the model config, initially I wanted to do it as a separate API and call it from the same caller site where ModelOperatorProvisioningInfo() is called, but wasn't sure if it was worth the overhead of an api call just to update a key in the model config. Was your suggestion in this comment "We should look at setting the correct model config value on model import as part of migration based on the content of the imported model", to put this update logic in the model migration stages instead i.e in func ImportModel(importer StateImporter,...)?